part-i-getting-started/messaging-concepts.md

```diff
@@ -102,3 +102,16 @@ When application components need resources at different stages of message proces
 
 When nested Units of Work need to be able to access a resource, it is recommended to register it on the root Unit of Work, which can be accessed using `unitOfWork.root()`. If a Unit of Work is the root, it will simply return itself.
 
+## Correlation Data Provider
+As already mentioned, Axon Framework is based on messaging concepts. In order to understand the flow of a living system, it is crucial to follow the message path. Hence, specific meta-data should be added to each message to be able to track its path through the system. Correlation Data is something that is usually added to a message its meta-data so that we can track the origin and causality of the message.
+
+Axon defines the `CorrelationDataProvider` as the mechanism to extract the Correlation Data from a given Message. There are three implementations of `CorrelationDataProvider` that come with Axon:
```

Comment:
Doesn't the CorrelationDataProvider add data to the message, instead of extracting it?